dateparser/conf.py

```diff
@@ -79,8 +81,133 @@ def wrapper(*args, **kwargs):
+            'values': (),
+            'type': str
+        },
+        'RETURN_AS_TIMEZONE_AWARE': {
+            # It defaults to 'default', but it's not allowed to use it directly
+            'values': (True, False),
+            'type': bool
+        },
+        'PREFER_DAY_OF_MONTH': {
+            'values': ('current', 'first', 'last'),
+            'type': str
+        },
+        'PREFER_DATES_FROM': {
+            'values': ('current_period', 'past', 'future'),
+            'type': str,
+        },
+        'RELATIVE_BASE': {
+            'values': (),  # unlimited options
+            'type': datetime
+        },
+        'STRICT_PARSING': {
+            'values': (True, False),
+            'type': bool
+        },
+        'REQUIRE_PARTS': {
+            'values': (),  # covered by the 'extra_check'
+            'type': list,
+            'extra_check': _check_require_part
+        },
+        'SKIP_TOKENS': {
+            'values': (),
+            'type': list,
+        },
+        'NORMALIZE': {
+            'values': (True, False),
+            'type': bool
+        },
+        'RETURN_TIME_AS_PERIOD': {
+            'values': (True, False),
+            'type': bool
+        },
+        'PARSERS': {
+            'values': (),  # covered by the 'extra_check'
+            'type': list,
+            'extra_check': _check_parsers
+        },
+        'FUZZY': {
+            'values': (True, False),
+            'type': bool
+        },
+        'PREFER_LOCALE_DATE_ORDER': {
+            'values': (True, False),
+            'type': bool
+        },
+    }
+
+    modified_settings = settings._mod_settings  # check only modified settings
+
+    # check settings keys:
+    for setting in modified_settings:
+        if setting not in settings_values:
+            raise SettingValidationError('"{}" is not a valid setting'.format(setting))
+
+    for setting_name, setting_value in modified_settings.items():
+        setting_type = type(setting_value)
+        setting_props = settings_values[setting_name]
+
+        # check type:
+        if not setting_type == setting_props['type']:
+            raise SettingValidationError(
+                '"{}" must be "{}", not "{}".'.format(
+                    setting_name, setting_props['type'].__name__, setting_type.__name__
+                )
+            )
+
+        # check values:
+        if setting_props['values'] and setting_value not in setting_props['values']:
+            raise SettingValidationError(
+                '"{}" is not a valid value for "{}", it should be: "{}" or "{}".'.format(
+                    setting_value,
+                    setting_name,
+                    '", "'.join(setting_props['values'][:-1]),
+                    setting_props['values'][-1],
+                )
+            )
+
+        # specific checks
+        extra_check = setting_props.get('extra_check')
+        if extra_check and not extra_check(setting_value):
```

Comment:
What if we let the check function raise the exception? That way, the message can be made more useful when possible.